### ISSUE TITLE
gpuR installation problem 

### ISSUE THREAD

**Issue by bobymathew** [cdeterman/gpuR#48]
I was trying to install gpuR in my ubuntu 16.10. I did the following to install the package   

devtools::install_github("cdeterman/RViennaCL")
devtools::install_github("cdeterman/gpuR", ref = "develop")

But I am getting a error 

/home/boby/R/x86_64-pc-linux-gnu-library/3.3/RViennaCL/include/viennacl/traits/size.hpp:164:44: error: ‘const class Eigen::Map<Eigen::Matrix<int, -1, -1>, 0,

/usr/lib/R/etc/Makeconf:141: recipe for target 'chol.o' failed
make: *** [chol.o] Error 1
ERROR: compilation failed for package ‘gpuR’

It would be great if you can help me in this regard.

Boby Mathew

**Comment by cdeterman** [comment]
Thanks for reporting the problem. I will need to set up a 16.10 virtual machine to see if I can reproduce. I don't think anyone has tried this on 16.10 yet.

**Comment by cdeterman** [comment]
@bobymathew what OpenCL platform are you using as well?  I would like to be as reproducible to your problem as possible.

**Comment by bobymathew** [comment]
I have nvidia. Many thanks.

**Comment by cdeterman** [comment]
@bobymathew How did you install the CUDA toolkit?  Did you install the 16.04 version?  Did you use the network or local files?  It isn't going particularly smoothly for me at the moment so if you can explain how you did it we can try to solve the problem faster.

**Comment by bobymathew** [comment]
I am using 16.10. I installed them using the synaptic package manager with ubuntu. Many thanks.

**Comment by cdeterman** [comment]
@bobymathew are you able to call `clinfo`?  I am having trouble getting the OpenCL to be found after installing:

```sh
sudo apt-get install nvidia-cuda-toolkit
```

**Comment by bobymathew** [comment]
When I call  clinfo I get

Number of platforms                               0

Boby

**Comment by bobymathew** [comment]
I have a warning like this when I try to install it 

/home/boby/R/x86_64-pc-linux-gnu-library/3.3/RViennaCL/include/viennacl/matrix.hpp:1186:38: note:   mismatched types ‘Eigen::Stride<0, 0>’ and ‘Eigen::OuterStride<>’
       assert( viennacl::traits::size2(cpu_matrix) == gpu_matrix.size2() && bool("Matrix dimensions mismatch: columns"));
 
May be this will help!

Boby

**Comment by cdeterman** [comment]
@bobymathew I don't know why you are receiving that error so I need to find a way to reproduce the problem first.  The program like `clinfo` should work and at the very least return 1 platform.  I suspect the NVIDIA toolkit install isn't working properly for 16.10.  I am trying to get this reproduced on a virtual machine but it doesn't have a NVIDIA GPU installed so it is difficult.  Until `clinfo` works properly I doubt `gpuR` will install regardless.

Another final test to make sure OpenCL is available and installed correctly is to use another one of repositories:

```sh
git clone https://github.com/cdeterman/intel_opencl.git
cd intel_opencl.git
g++ clDeviceQuery.cpp -o clDeviceQuery -lOpenCL
./clDeviceQuery
```

**Comment by bobymathew** [comment]
Now I get this error  when I run ./clDeviceQuery

clDeviceQuery Starting...

 Error -1001 in clGetPlatformIDs Call!

Boby

**Comment by cdeterman** [comment]
@bobymathew exactly, that's my point.  The NVIDIA OpenCL platform is not being found on your OS.  I'm still trying to figure out why this is happening on 16.10.

**Comment by cdeterman** [comment]
@bobymathew okay, well I can at least get gpuR to compile although the package cannot load because the OpenCL driver is still not happy.  Basically I set the environmental variable `CUDA_HOME` and ran the install:

```r
Sys.setenv(CUDA_HOME = "/usr/local/cuda-8.0/targets/x86_64-linux/lib")
devtools::install_gitub('cdeterman/gpuR', ref = 'develop')
```

**Comment by bobymathew** [comment]
Many thanks.. May be I will wait until someone fix this issue of OpenCL with ubuntu 16.10.
Boby

**Comment by cdeterman** [comment]
@bobymathew I actually opened a [question](https://devtalk.nvidia.com/default/topic/975944/cuda-setup-and-installation/ubuntu-16-10-cuda-toolkit-install-with-opencl/?offset=5#5015945) with NVIDIA and it appears that the CUDA toolkit is just not supported yet on 16.10.  Will need to wait and see when this support comes.

**Comment by bobymathew** [comment]
Many many thanks and I wait for it.

**Comment by anadon** [comment]
@bobymathew Checking in at the following:
https://launchpad.net/ubuntu/yakkety/amd64/nvidia-cuda-toolkit
https://devtalk.nvidia.com/default/topic/975944/cuda-setup-and-installation/ubuntu-16-10-cuda-toolkit-install-with-opencl/

It is likely that you have a broken driver installation.  Speaking from past systems, the closed source AMD and Nvidia drivers can be very fragile.
